test/test_constrained_dataset.py

```diff
@@ -0,0 +1,74 @@
+import unittest
+import numpy as np
+from metric_learn.constraints import ConstrainedDataset
+from numpy.testing import assert_array_equal
+from sklearn.model_selection import StratifiedKFold, KFold
+from sklearn.utils.testing import assert_raise_message
+
+X = np.random.randn(20, 5)
```

Comment:
It would be nice to test sparse `X` as well. To do so, I'd write something like this:
```python
class _BaseTestConstrainedDataset(unittest.TestCase):
   # everything currently under TestConstrainedDataset, but using self.X instead of X

class TestDenseConstrainedDataset(_BaseTestConstrainedDataset):
  def setUp(self):
    self.X = np.random.randn(20, 5)
    self.c = ... # and so on

class TestSparseConstrainedDataset(_BaseTestConstrainedDataset):
   # similar, but setUp creates a dense X
```

Reply:
maybe it would make sense to also test other data types? like lists instead of numpy arrays

Reply:
I agree, I will add these tests in a next commit